app/validation/validator.py

```diff
@@ -193,123 +172,86 @@ def _get_numeric_range_values(self, answer, answer_ranges):
 
-                    if option['value'] in values:
-                        errors.append(self._error_message('Duplicate value found - {}'.format(option['value'])))
-
-                    labels.add(option['label'])
-                    values.add(option['value'])
+            labels.add(option['label'])
+            values.add(option['value'])
 
         return errors
 
-    def validate_contains_confirmation_or_summary(self, json_to_validate):
-        blocks = self._get_blocks(json_to_validate)
-        for block in blocks:
-            if block['type'] in ['Summary', 'Confirmation']:
-                return []
+    def validate_contains_confirmation_or_summary(self, last_block):
+        if last_block['type'] in ['Summary', 'Confirmation']:
+            return []
 
         return [self._error_message('Schemas does not have a confirmation or summary page')]
 
-    def validate_child_answers_define_parent(self, json_to_validate):
+    def validate_child_answers_define_parent(self, answers):
         errors = []
 
-        for block in self._get_blocks(json_to_validate):
-            answers_by_id = self._get_answers_by_id_for_block(block)
-
-            for answer_id, answer in answers_by_id.items():
-                if answer['type'] in ['Radio', 'Checkbox']:
-                    child_answer_ids = (o['child_answer_id'] for o in answer['options'] if 'child_answer_id' in o
-                                        and o['child_answer_id'])
-
-                    for child_answer_id in child_answer_ids:
-                        if child_answer_id not in answers_by_id:
-                            errors.extend([self._error_message('Child answer with id %s does not exist in schemas'
-                                                               % child_answer_id)])
-                            continue
-                        if 'parent_answer_id' not in answers_by_id[child_answer_id]:
-                            errors.extend([self._error_message('Child answer %s does not define parent_answer_id %s '
-                                                               'in schemas' % (child_answer_id, answer_id))])
-                            continue
-                        if answers_by_id[child_answer_id]['parent_answer_id'] != answer_id:
-                            errors.extend([self._error_message('Child answer %s defines incorrect parent_answer_id %s '
-                                                               'in schemas: Should be %s'
-                                                               % (child_answer_id,
-                                                                  answers_by_id[child_answer_id]['parent_answer_id'],
-                                                                  answer_id))])
-                            continue
+        answers_by_id = self._get_answers_by_id_for_block(answers)
+
+        for answer in answers:
+            if answer['type'] in ['Radio', 'Checkbox']:
```

Comment:
Again not sure if these would be better served being above before coming into this method

Reply:
I think its better to keep it contained in the function